code/game/objects/items/binoculars.dm

```diff
@@ -134,16 +146,19 @@
 	if(!changable)
 		to_chat(user, "These binoculars only have one mode.")
 		return
-	mode += 1
-	if(mode > MODE_ORBITAL)
-		mode = MODE_CAS
+	var/n_mode = modes.Find(mode)
+	mode = modes[n_mode + 1]
+	if(mode > modes.len)
+		mode = modes[1]
```

Comment:
I prefer the old code IMO

Reply:
I was trying to make it a bit simpler for adding binocs with different types, this code doesnt work atm though, so i might give up